SYCL/ESIMD/api/functional/operators/operator_decrement_and_increment.hpp

```diff
@@ -0,0 +1,254 @@
+//===-- operator_decrement_and_increment.hpp - Functions for tests on simd
+//      increment and decrement operators definition. ---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd decrement and increment
+/// operators.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+#include "../mutator.hpp"
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+namespace esimd_functional = esimd_test::api::functional;
+
+namespace esimd_test::api::functional::operators {
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct pre_decrement {
+  static std::string get_operator_type() { return " pre decrement"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(const DataT *const ref_data,
+                             DataT *const source_simd_out,
+                             DataT *const result_simd_out) {
+    auto source_simd = esimd::simd<DataT, NumElems>();
+    source_simd.copy_from(ref_data);
+    esimd::simd<DataT, NumElems> result_simd = --source_simd;
+    source_simd.copy_to(source_simd_out);
+    result_simd.copy_to(result_simd_out);
+  }
+
+  template <typename DataT> static DataT apply_operator(DataT val) {
+    return --val;
+  }
+
+  static constexpr bool is_increment() { return false; }
+};
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct post_decrement {
+  static std::string get_operator_type() { return " post decrement"; }
```

Comment:
Looks like `type` suffix is misleading here.

I would suggest either to use the same name as for all other test cases
```suggestion
  static std::string get_description() { return "post decrement"; }
```
or to use `named_type_pack` alongside with
```suggestion
  static std::string to_string() { return "post decrement"; }
```

Reply:
this comment was applied in this commit: d0f0c3a641f23798485b102a95bc802b53957700